### DISCUSSION TITLE
Persist URL parameters during language switch

### DISCUSSION THREAD

**Discussion by digital-codes** [vuepress/core#1204] [Q&A] [answered]
Hello. I use URL parameters to select dynamic content to be loaded into a static page, for example issues of a newsletter or blog. Well, this might be beyond the scope of a static website builder, but it's a nice feature. My solution (at https://github.com/digital-codes/vp2, demo e.g. https://vp2.akugel.de/news.html?id=1  ) works in principle. However, when I switch language I loose the URL parameters and bounce at the empty static page. Would it be possible to preserve the URL parms across language switching e.g. with a route-guard? I assume lang switch is part of the default-theme ....

**Comment by Mister-Hope** [comment, marked as answer]
https://github.com/vuepress/vuepress-next/pull/1211

**Reply by digital-codes** [reply]
Thanks for looking into this. I applied your change to NavBarItems.vue locally, looks good from my side!
